server/webapp/WEB-INF/rails.new/webpack/models/notifications/data_sharing_notification_permissions.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const AjaxHelper = require('helpers/ajax_helper');
+const Stream                          = require('mithril/stream');
+
+const DataSharingNotificationPermission = function (permission) {
+    const self = this;
+    this.shouldAllow = Stream(permission.allow);
+};
+DataSharingNotificationPermission.API_VERSION = 'v1';
+
+DataSharingNotificationPermission.get = () => {
+    return AjaxHelper.GET({
+        url:        "/go/api/data_sharing/settings/notification",
```

Comment:
Move the URL definition to `SparkRoute`?

Reply:
I thought `SparkRoute` was some fancy way of accessing the spark URLs, so I was trying to figure out how does it work. Was disappointed to see that its just hard-coded urls in there. Didn't see any use for it, so ditched that.

Reply:
@jyotisingh -- We wanted to build something similar to `js-routes`. But unfortunately, there is nothing available which can do that for us.

The motive behind defining URLs in sparkRoute is to have them at one place, so if at the later point in time the endpoint URL changes, we just have to update the URL only in one place.